client/src/main/kotlin/io/spine/examples/shareaware/client/MarketPage.kt

```diff
@@ -297,63 +337,478 @@ public class MarketPageModel(client: DesktopClient) {
-                        containerColor = MaterialTheme.colorScheme.tertiary
-                    )
+                        .size(22.dp)
+                        .padding(end = 5.dp)
                 )
-                Divider(
-                    thickness = 2.dp
+            }
+        )
+    }
+}
+
+/**
+ * Represents the list of shares.
+ *
+ * @param listState the state object to be used to control or observe the list's state
+ * @param shares the list of the actual market shares
+ * @param previousShares the list of the previous market shares
+ * @param model the model of the market page
+ */
+@Composable
+private fun SharesList(
+    listState: LazyListState,
+    shares: List<Share>,
+    previousShares: List<Share>?,
+    model: MarketPageModel
+) {
+    LazyColumn(
+        modifier = Modifier.fillMaxSize(),
+        state = listState
+    ) {
+        shares.forEachIndexed { index, share ->
+            item {
+                ShareItem(
+                    model = model,
+                    share = share,
+                    previousPrice = previousShares?.get(index)?.price
                 )
             }
-            val purchaseInProgress = model.purchaseOperation.isInProgress().collectAsState()
-            PurchaseDialog(
-                model = model,
-                isShown = purchaseInProgress.value
+        }
+    }
+}
+
+/**
+ * Represents an empty shares list.
+ */
+@Composable
+private fun EmptySharesList() {
+    Column (
+        modifier = Modifier
+            .fillMaxSize(),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Text(
+            text = "Nothing to show",
+            style = MaterialTheme.typography.bodySmall,
+            color = MaterialTheme.colorScheme.onSecondary
+        )
+    }
+}
+
+
+/**
+ * Represents the list item with information about the share.
+ *
+ * @param model the model of the "Market" page
+ * @param share the share to represent
+ * @param previousPrice the previous price of this share
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun ShareItem(
+    model: MarketPageModel,
+    share: Share,
+    previousPrice: Money?
+) {
+    Box(
+        modifier = Modifier
+            .clip(MaterialTheme.shapes.small)
+            .bottomBorder()
+    ) {
+        ListItem(
+            modifier = Modifier
+                .height(60.dp)
+                .clickable(
+                    enabled = true,
+                    onClick = {
+                        model.selectedShare(share.id)
+                    }
+                ),
+            headlineText = {
+                MainItemContent(share, previousPrice)
+            },
+        )
+    }
+}
+
+/**
+ * Represents the main `ListItem` content with data about the share.
+ *
+ * @param share the share to show inside the item
+ * @param previousPrice the previous price of this share
+ */
+@Composable
+private fun MainItemContent(
```

Comment:
I think it's better to rename `MainItemContent` to `ShareItemContent` because it strongly depends on the `Share` and corresponds to the other naming.